### PR TITLE
Use NanostackRfPhy::get_default_instance() for finding the driver

### PR DESCRIPTION
**Pull request by SeppoTakalo**
### Description

Use NanostackRfPhy::get_default_instance() for finding the driver

@JuhPuur  please test & review.

### CODE REVIEW

**Comment by SeppoTakalo** [comment]
For this to work, following changes are required for Mbed OS https://github.com/ARMmbed/mbed-os/pull/7778 or altearnatively update Atmel and MCR20A driver to master.

Then, without changes, you should be able to compile for NCS36510.
Or KW24D when MCR20A driver is added.

For compiling to Atmel RF driver, you need to follow the README.md and add `DEVICE_802_15_4_PHY` flag.

File: TEST_APPS/device/nanostack_mac_tester/main.cpp [hunk 1]
```diff
@@ -94,6 +87,7 @@ static void handle_rx_data(void)
 
 static int mac_prepare(void)
 {
+    NanostackRfPhy rf_phy = NanostackRfPhy::get_default_instance();
```
Comment (JuhPuur) [inline TEST_APPS/device/nanostack_mac_tester/main.cpp:90]: NanostackRfPhy &rf_phy = NanostackRfPhy::get_default_instance();

File: TEST_APPS/device/nanostack_mac_tester/main.cpp [hunk 2]
```diff
@@ -33,17 +33,10 @@
 #define TRACE_GROUP "Main"
 #include "mbed-trace/mbed_trace.h"
```
Comment (JuhPuur) [inline TEST_APPS/device/nanostack_mac_tester/main.cpp:35]: #include "NanostackRfPhy.h"

**Comment by JuhPuur** [comment]
Compiled succesfully with atmel driver and tested succesfully. With those small changes this can be merged. There are a few things that need to be fixed but I'd rather shorten the worrying string of pull requests(3 deep now) before doing more work.

**Pull request merged**